extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -120,56 +158,42 @@ void TailFile::onSchedule(const std::shared_ptr<core::ProcessContext> &context,
   std::string value;
 
   if (context->getProperty(Delimiter.getName(), value)) {
-    delimiter_ = value;
+    delimiter_ = parseDelimiter(value);
+  }
+
+  if (!context->getProperty(FileName.getName(), file_to_tail_)) {
+    throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "File to Tail is a required property");
   }
 
   std::string mode;
   context->getProperty(TailMode.getName(), mode);
 
-  std::string file = "";
-  if (!context->getProperty(FileName.getName(), file)) {
-    throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "File to Tail is a required property");
-  }
   if (mode == "Multiple file") {
-    // file is a regex
-    std::string base_dir;
-    if (!context->getProperty(BaseDirectory.getName(), base_dir)) {
+    tail_mode_ = Mode::MULTIPLE;
+
+    if (!context->getProperty(BaseDirectory.getName(), base_dir_)) {
       throw minifi::Exception(ExceptionType::PROCESSOR_EXCEPTION, "Base directory is required for multiple tail mode.");
```

Comment:
This is fair, but I would also add a check here to make sure that the directory exists. 
As far as I remember we have such utility func in fileutils, so not a big change.

Reply:
hm, not sure...  do we want to refuse to start if the directory does not exist (yet)?  maybe it will be created later

Reply:
OnSchedule throwing an exception results in being called after a while (30s by default as far as I remember).

So that doesn't mean giving it up completely, just trying later and only let ontrigger happen when at least we can list the dir.

Reply:
done